sonarqube/resource_sonarqube_qualitygate.go

```diff
@@ -129,6 +178,12 @@ func resourceSonarqubeQualityGateRead(d *schema.ResourceData, m interface{}) err
 	d.Set("name", qualityGateReadResponse.Name)
 	// Api returns if true if set as default is available. when is_default=true setAsDefault=false so is_default=tue
 	d.Set("is_default", !qualityGateReadResponse.Actions.SetAsDefault)
+
```

Comment:
Is `d.Set("conditions", qualityGateReadResponse.Conditions)` not suficient? 

I dont understand why we need to remarshal this back into json? If we really need this in json, can just use resp.Body? I think that should already be in json.

Reply:
I thought so too, but It's not. Using `d.Set("conditions", qualityGateReadResponse.Conditions)` produces this error:
![bild](https://user-images.githubusercontent.com/34520175/221495751-a514487a-8ffb-4a16-a852-2fecba6e337b.png)

I'm not the most adverse go-coder, so I can't explain exactly why, but using the below snippet works. 
```go
var allConditions []map[string]string
data, _ := json.Marshal(qualityGateReadResponse.Conditions)
json.Unmarshal(data, &allConditions)
d.Set("conditions", allConditions)
// d.Set("conditions", qualityGateReadResponse.Conditions)
```

Reply:
I can dig into it more after work though

Reply:
I based my solution for the schema on the Terraform provider for GitHub by the way, as seen [here](https://github.com/integrations/terraform-provider-github/blob/b4042f1c704d29f03f9619488a89e085b0b17b96/github/data_source_github_actions_organization_secrets.go#L15-L40)

Reply:
Thanks for the troubleshooting and in-depth explanation.